test/unit/test_test_helpers.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+import jubilant
+
+from . import mocks
+
+
+def mock_token_hex(n: int):
+    assert n == 4
+    return 'abcd1234'
+
+
+def test_defaults(run: mocks.Run, monkeypatch: pytest.MonkeyPatch):
+    monkeypatch.setattr('secrets.token_hex', mock_token_hex)
+    run.handle(['juju', 'add-model', '--no-switch', 'jubilant-abcd1234'])
+    run.handle(['juju', 'deploy', '--model', 'jubilant-abcd1234', 'app1'])
+    run.handle(
+        [
+            'juju',
+            'destroy-model',
+            'jubilant-abcd1234',
+            '--no-prompt',
+            '--destroy-storage',
+            '--force',
+        ]
+    )
+
+    with jubilant.temp_model() as juju:
+        assert juju.model == 'jubilant-abcd1234'
+        juju.deploy('app1')
+
+    assert juju.model is None
+    assert len(run.calls) == 3
+    assert run.calls[0].args[1] == 'add-model'
+    assert run.calls[1].args[1] == 'deploy'
```

Comment:
It makes the arrange/act/assert less clean, but it feels like the test ought to assert that the add and deploy were done when you first enter the context and the destroy is done after exiting.

Reply:
Ah yes, good call! Fixed.